packages/rs-dpp/src/data_contract/document_type/document_factory.rs

```diff
@@ -0,0 +1,49 @@
+use crate::data_contract::document_type::property_names::{CREATED_AT, UPDATED_AT};
+use crate::data_contract::document_type::DocumentType;
+use crate::document::document_transition::INITIAL_REVISION;
+use crate::document::Document;
+use crate::prelude::TimestampMillis;
+use crate::ProtocolError;
+use chrono::Utc;
+use platform_value::Value;
+
+use platform_value::Identifier;
+use std::collections::BTreeMap;
+
+impl DocumentType {
+    /// Creates a document at the current time based on document type information
+    /// Properties set here must be pre validated
```

Comment:
Why?

Reply:
Maybe I don't understand something here, but why document will have no created date? Why would we permit it to be missed from properties list? And why it would create a document with valid properties only because it has some assumption that input properties are actually valid? it's just arbitrary btreemap; moreover this is not some local function, even `pub`;

either this docstring confused me or we're missing a single construction point of valid `Document`